scripts/install_configuration.yaml

```diff
@@ -0,0 +1,41 @@
+system_requirements_mac:
```

Comment:
I like this idea, but it currently fails the single source of truth test in that the install script doesn't read this file, so this doesn't actually fix the problem of the install script changing and the documentation failing to keep up. 

This is probably me being unclear. What I had in mind is that the documentation would run the install script passing appropriate options that will make the install script print out what it would have installed. For example `firedrake-install --show-petsc-configure-options` already does this for PETSc options, and `firedrake-install --show-dependencies` very nearly does what is needed for packages. I don't think there is currently a way of outputting the wheel blacklist, but it wouldn't be too hard.

Reply:
Yes, but building then the documentation will take a lot of time or am I missing the point. (we currently just have the install and update mode). And the second problem would be the difference between the mac and linux configuration.

Reply:
If you run the install script with the options provided, it prints out the information required and exits immediately. It doesn't run an actual install.

The difference between Mac and Linux is what I was referring to when I said that `firedrake-install --show-dependencies` very nearly does what is needed. I suggest adding an optional argument to the `--show-dependencies` flag to enable the platform to be specified (instead of being detected as it is now).

Reply:
I should also clarify that the issue with having the install script read the yaml file is that the install script gets downloaded and used separately from the git repository. There is no way for the install script to know which version of the yaml file to use. This is why it's important that the install script itself is the single source of truth, and that the documentation queries the install script.

Reply:
Ok. I understand. I also thought that you want a singular file that should do all the work. Currently the why I have managed to introduce dynamic content is by creating .rst files. Should I run the install script and create this .rst files? I think this should be a good solution.